moveit_core/robot_model/include/moveit/robot_model/link_model.h

```diff
@@ -41,13 +41,16 @@
 #include <vector>
 #include <utility>
 #include <map>
+#include <boost/shared_ptr.hpp>
 #include <Eigen/Geometry>
 #include <eigen_stl_containers/eigen_stl_vector_container.h>
 #include <moveit/macros/class_forward.h>
 
 namespace shapes
 {
-MOVEIT_CLASS_FORWARD(Shape);
+class Shape;
+typedef boost::shared_ptr<Shape> ShapePtr;
+typedef boost::shared_ptr<const Shape> ShapeConstPtr;
```

Comment:
Hmm, changing the type of `geometric_shapes::ShapePtr` is not really a nice solution. Best is if `geometric_shapes` dictates the type of `shared_ptr`, but that is not going to happen without including a `geometric_shapes` header.

So that leaves pretty much just two options:
* Add a forward declarations header to `geometric_shapes`
* Revert the largest part of 13d2e622658c995e140f85a08891ea65ba8d7dd5

@v4hn, @rhaschke: Thoughts?